### PR TITLE
Front end js

### PR DESCRIPTION
**Pull request by jpc20**
Index pages are now divided into two parts(NYTimes and Community) and can be switched back and forth with a button.
Updated tests for the index pages to pass with this new setup.
Also added some front-end bug fixes.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -40,6 +40,8 @@ group :development, :test do
   gem 'faker'
   gem 'pry-rescue'
   gem 'pry-stack_explorer'
+  gem 'selenium-webdriver'
+  gem 'webdrivers'
```
Comment (jpc20) [inline Gemfile:44]: Added these so the javascript would run in RSpec.

File: app/assets/stylesheets/application.scss [hunk 2]
```diff
@@ -213,6 +213,10 @@ body {
   transition: all 0.2s;
 }
 
+.hide-dom {
+  display: none;
+}
```
Comment (jpc20) [inline app/assets/stylesheets/application.scss:218]: I am using this CSS class to hide the index partial that the user has not selected.

File: app/poros/api_obituary.rb [hunk 3]
```diff
@@ -23,6 +23,6 @@ def format_date(date)
   end
 
   def format_image(url)
-    "https://static01.nyt.com/".concat(url)
+    url ? "https://www.nytimes.com/".concat(url) : 'https://d1yjjnpx0p53s8.cloudfront.net/styles/logo-thumbnail/s3/0013/5263/brand.gif?itok=MkqjUMfP'
```
Comment (jpc20) [inline app/poros/api_obituary.rb:26]: This adds a default image if the NYTimes doesn't return one. I just chose a random image and it isn't formatted the same as the NYTimes thumbnails so we will need to style this, but I needed it to stop the page from throwing an error.

File: app/views/application/_browsing_index.html.erb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    </div>
+    <div class="col-md-9 p-4">
+      <h3><%= "#{obit.first_name} #{obit.last_name}" %></h3>
+      <p><%= obit.description %></p>
+      <a class="btn btn-dark" href= <%= "/obituaries/#{obit.id}" %>>View Obituary</a>
+    </div>
+  </div>
+  <hr>
+<% end %>
+</div>
```
Comment (jpc20) [inline app/views/application/_browsing_index.html.erb:15]: @stellakunzang 's partial!

File: app/views/obituaries/covid/index.html.erb [hunk 5]
```diff
@@ -1,39 +1,27 @@
...
-      </div>
-    </section>
-  <% end %>
 </div>
+
+<script type="text/javascript">
+  $('#change').click(function() {
+     $('#community-results, #nytimes-results').toggle();
+    $(this).prop('value') == 'View NYTimes Results' ? communityText() : nytimesText();
+  });
```
Comment (jpc20) [inline app/views/obituaries/covid/index.html.erb:16]: This changes the text on the button, the header, and the partial being rendered when it is clicked.

File: app/views/obituaries/show.html.erb [hunk 6]
```diff
@@ -31,7 +31,7 @@
       <br>
       <% if current_user && current_user.has_obituary?(@obituary) %>
         <button class="btn btn-outline-secondary mr-1"><%= link_to 'Edit', edit_obituary_path(@obituary.id) %></button>
-        <button class="btn btn-outline-secondary mr-1"><%= link_to 'Delete', "/obituaries/#{@obituary.id}", method: :delete %></button>
+        <button class="btn btn-outline-secondary mr-1"><%= link_to 'Delete', "/obituaries/#{@obituary.id}", method: :delete, data: { confirm: 'Are you sure you want to delete this obituary?' } %></button>
```
Comment (jpc20) [inline app/views/obituaries/show.html.erb:34]: Adds a browser alert asking the user to confirm when they are trying to delete.

File: spec/rails_helper.rb [hunk 7]
```diff
@@ -8,6 +8,16 @@
 # Prevent database truncation if the environment is production
 abort("The Rails environment is running in production mode!") if Rails.env.production?
 require 'rspec/rails'
+Capybara.register_driver :selenium do |app|
+  options = Selenium::WebDriver::Chrome::Options.new(args: %w[no-sandbox headless disable-gpu])
+  Capybara::Selenium::Driver.new(app, browser: :chrome, options: options)
+end
```
Comment (jpc20) [inline spec/rails_helper.rb:14]: Set up for getting javascript to work in rspec

File: app/views/obituaries/covid/index.html.erb (hunk 5 above)
Comment (EricLarson2020) [reply app/views/obituaries/covid/index.html.erb:16]: Nice work here!

File: app/views/application/_nytimes_index.html.erb [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    </thead>
+    <tbody style= "background-color:white;">
+      <% @obituaries_api.each do |obit| %>
+        <tr id="obituary_by_name">
+          <td id="image"><%= image_tag(obit.image) %></td>
+          <td id="headline" class="align-middle pl-3"><%= obit.headline %></td>
+          <td id="pub_date" class="align-middle pl-3"><%= obit.pub_date %></td>
+          <td id="abstract" class="align-middle pl-3"><%= obit.abstract %></td>
+          <td id="word_count" class="align-middle pl-3"><%= obit.word_count %></td>
+          <td id="web_url" class="align-middle pl-3"><%= link_to obit.web_url, obit.web_url %></td>
```
Comment (jpc20) [inline app/views/application/_nytimes_index.html.erb:20]: This url should be working fine now. We just had to put it as the path as well. If we have time we can think about a  better way to link each listing to the actual article rather that the raw url.

File: app/poros/api_obituary.rb (hunk 3 above)
Comment (sagemlee) [reply app/poros/api_obituary.rb:26]: Good problem solving.

File: app/views/application/_browsing_index.html.erb (hunk 4 above)
Comment (sagemlee) [reply app/views/application/_browsing_index.html.erb:15]: What a beaut

File: app/views/obituaries/advanced/index.html.erb [hunk 9]
```diff
@@ -1,23 +1,4 @@
...
-    </div>
-    <div class="word_count">
-      <%= obit.word_count %>
-    </div>
-    <div class="image">
-      <%= obit.image %>
-    </div>
-  </section>
-<% end %>
+<div class="container">
```
Comment (sagemlee) [inline app/views/obituaries/advanced/index.html.erb:1]: Nice, this looks way cleaner

File: app/views/obituaries/covid/index.html.erb (hunk 5 above)
Comment (sagemlee) [reply app/views/obituaries/covid/index.html.erb:16]: Javascript is magic.

File: app/views/obituaries/covid/index.html.erb [hunk 10]
```diff
@@ -1,39 +1,27 @@
...
+
+  function communityText() {
+    $('#change').prop('value', 'View Community Results')
+    $('#results-header').text('COVID Related NYTimes Obituaries:')
+  }
+
+  function nytimesText() {
```
Comment (sagemlee) [inline app/views/obituaries/covid/index.html.erb:23]: You can start to see a pattern with how function is similar to define in ruby, but I hate this "()" nonesense.

File: config/environments/production.rb [hunk 11]
```diff
@@ -91,7 +91,7 @@
 
   config.action_mailer.perform_deliveries = true
   config.action_mailer.smtp_settings = {
-    domain: 'https://obituary-hub.herokuapp.com/,
+    domain: 'https://obituary-hub.herokuapp.com/',
```
Comment (sagemlee) [inline config/environments/production.rb:94]: wow! nice eye, would have taken me forever to find this quote.

File: app/views/obituaries/covid/index.html.erb (hunk 10 above)
Comment (jpc20) [reply app/views/obituaries/covid/index.html.erb:23]: Same! There is way too much punctuation lol

File: config/environments/production.rb (hunk 11 above)
Comment (sagemlee) [reply config/environments/production.rb:94]: Hopefully we are up and running once this is changed

File: spec/rails_helper.rb (hunk 7 above)
Comment (sagemlee) [reply spec/rails_helper.rb:14]: Glad you were able to figure out testing! This looks complicated.

**Comment by sagemlee** [comment]
This looks great Jack! You added a lot of code, I can tell you put a lot of work into it.

File: app/assets/stylesheets/application.scss (hunk 2 above)
Comment (stellakunzang) [reply app/assets/stylesheets/application.scss:218]: Nice!

File: app/poros/api_obituary.rb (hunk 3 above)
Comment (stellakunzang) [reply app/poros/api_obituary.rb:26]: THANK YOU YAAAAYYYYYY

File: app/views/obituaries/advanced/index.html.erb (hunk 9 above)
Comment (stellakunzang) [reply app/views/obituaries/advanced/index.html.erb:1]: Definitely cleaner!

**Pull request merged**